platform/core/src/redux/reducers/viewports.js

```diff
@@ -25,6 +25,40 @@ const DEFAULT_STATE = {
   viewportSpecificData: {},
 };
 
+/**
+ *  Take the new number of rows and columns, delete all not used viewport data and also set
+ *  active viewport as default in case current one is not available anymore.
+ *
+ * @param {Object} action -
+ * @param {Number} action.numRows -
+ * @param {Number} action.numColumns -
+ * @param {Object} currentViewportSpecificData -
+ * @param {Number} currentActiveViewportIndex -
+ * @returns
+ */
+const handleViewportDeletions = (action, currentViewportSpecificData = {}, currentActiveViewportIndex) => {
```

Comment:
`handle` is a good word for handling events, but this little guy is finding/determining our viewports given the number of viewports.

We should change its name, update the parameters to take the values it needs from `action` more explicitly, and update/add unit tests for the reducers that use this helper to assert that this expected behavior happens.